core/src/test/kotlin/generator/TestRunner.kt

```diff
@@ -0,0 +1,101 @@
+package generator
+
+import org.jetbrains.dokka.*
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.jetbrains.dokka.utilities.DokkaLogger
+import java.io.File
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+
+object TestRunner {
+    private var testDir: Path? = null
+
+    fun generateTestDokka(
+        name: String,
+        query: String,
+        checks: Map<DokkaTestGenerator.Companion.Stage, (DokkaTestGenerator.State) -> Boolean>,
+        targets: List<String> = listOf("jvm"),
+        platforms: List<String> = listOf("jvm"),
+        logger: DokkaLogger = DokkaConsoleLogger
+    ) {
+        if (testDir == null) {
+            testDir = Files.createTempDirectory("tests")
+            testDir!!.toFile().deleteOnExit()
+        }
+
+        val root = testDir!!.resolve(name)
+        root.deleteContents()
+
+        val output = root.resolve("out")
+        Files.createDirectories(output)
+
+        platforms.mapNotNull { it.toPlatform() }
+            .forEach {
+                output.deleteContents()
+
+                val config = DokkaConfigurationImpl(
+                    outputDir = output.toString(),
+                    format = "html",
+                    generateIndexPages = true,
+                    passesConfigurations = listOf(
+                        PassConfigurationImpl(
+                            moduleName = name,
+                            sourceRoots = listOf(root).map { SourceRootImpl(it.toString()) },
+                            analysisPlatform = it,
+                            targets = targets
+                        )
+                    )
+                )
```

Comment:
Still no way to change config other than writing a new `generateTestDokka` method